iso-configuration/default.nix

```diff
@@ -3,22 +3,23 @@
 {
   imports = [
     ./installer-configuration.nix
-    ../m1-support
+    ../nixos-module
   ];
 
   # include those modules so the user can rebuild the install iso. that's not
   # especially useful at this point, but the user will need the m1-support
   # directory for their own config.
   installer.cloneConfigIncludes = [
     "./installer-configuration.nix"
-    "./m1-support"
+    "./m1-support/nixos-module"
   ];
 
   # copy the m1-support and installer configs into the iso
   boot.postBootCommands = lib.optionalString config.installer.cloneConfig ''
     if ! [ -e /etc/nixos/m1-support ]; then
+      mkdir -p /etc/nixos/m1-support
       cp ${./installer-configuration.nix} /etc/nixos/installer-configuration.nix
-      cp -r ${../m1-support} /etc/nixos/m1-support
+      cp -r ${../packages} ${../nixos-module} -t /etc/nixos/m1-support
```

Comment:
This needs to copy each module into the `m1-support` folder by name directly. Currently the store hashes get included in the path name so the user will end up with a bunch of copies of the two folders instead of overwriting the old one.

We also need a `default.nix` which simply includes the module directory so that the user can just add the `m1-support` directory to their config.

Alternatively I would prefer to migrate both the `nixos-module` and `packages` folders into an `m1-support` folder in the repo (and add that `default.nix` so that channel-based upgrades and people with it checked out would continue to work.

Reply:
> Currently the store hashes get included in the path name so the user will end up with a bunch of copies of the two folders instead of overwriting the old one.

What do you mean by this?

My intent with the installer command was to preserve the project structure so that relative paths (like `../packages/overlay.nix`) would work, and so that channel or tarball setups would need minimal changes.

The new "module usage instructions" would be to import `m1-support/nixos-module` instead of `m1-support`.

If you just want to make sure that existing setups won't break, we could make a trivial `m1-support/default.nix`, maybe with a deprecation warning. (Is this what you meant in the second paragraph?)

My reasoning in keeping `nixos-module` (formerly `m1-support`) and `packages` separate was that `packages` is sort of independent from our nixos-module. It defines our overlay, which is used for both the nixos module and for flake outputs (m1n1 and u-boot), and it makes sense outside of the context of nixos configurations.

Reply:
The problem is that right now, in the installer, there are folders `/etc/nixos/m1-support/<random hash>-packages` and `/etc/nixos/m1-support/<random hash>-nixos-module` so the relative paths are broken. You have to specify the destination explicitly in `cp` to prevent this and copy the two directories in two separate commands.

I agree with the division into two separate folders. But right now it is `<repo>/iso-configuration`, `<repo>/nixos-module` and `<repo>/packages`. I would like it to be `<repo>/iso-configuration`, `<repo>/m1-support/nixos-module` and `<repo>/m1-support/packages`. That way only one folder has to be copied, both in the installer above and by the user to upgrade. That folder contains everything necessary for the module to work.

Additionally, I would like `<repo>/m1-support/default.nix` to be a simple module which just imports `<repo>/m1-support/nixos-module`. I don't think a deprecation warning is necessary, I can't think of a reason to remove this file. This is also necessary for a user who copies the new project structure over the old one to get correct operation and use the new folders.

Reply:
(Another thing to consider is that we might want to add "M2 support" pretty soon, and we'd want a good directory structure in place for that.)

Reply:
Another thing to consider is that this repo is for Apple silicon support in general, and I think we should be moving away from referencing "M1," especially with M2 already released.

Reply:
Maybe we could use `nixos-module/modules` and `nixos-module/packages`?

Reply:
What about `as-support/modules` and `as-support/packages`? `nixos-as` is a bit generic though but it's the best I have. I don't think it would be appropriate to call it `nixos-asahi`.

Reply:
Maybe `apple-silicon-support` or `nixos-apple-silicon`?

Why wouldn't `nixos-asahi` work? [Asahi Linux's about page](https://asahilinux.org/about/) describes itself as "an overall project to develop Linux support for Apple Silicon Macs," and it happens to support an Arch distro under the same name.

If all of Asahi gets upstreamed, then Asahi will no longer exist, and this repo would not need to exist.

Reply:
I like `apple-silicon-support`, with subdirectories `modules` and `packages`, and the `default.nix`.

Regarding Asahi, while I use their work and credit them (and even contribute a little), I don't want to use their name for my project. I don't want to misrepresent that this is officially created or sanctioned or supported by them.

Reply:
I've pushed a commit with `apple-silicon-support`.